bootstrap_salt/fab_tasks.py

```diff
@@ -145,7 +145,41 @@ def create_kms_data_key():
     return kms.generate_data_key(get_kms_key_id())
 
 
-bcfn_create, bcfn_delete = cfn_create, cfn_delete
+@task
+def get_kms_data_key():
+    """
+    Returns a decrypted version of the current data key.
+
+    The current data key is needed, when we need to reconstruct the
+    LaunchConfiguration for an existing stack.
+
+    XXX: maybe we could add data key rotation here, although this
+    would break things before the next vendor_upload.
+    """
+    try:
+        ips = get_instance_ips()
+    except:
+        ips = None
+
+    #
+    # If we don't currently have any instances just create a new key
+    #
+    # XXX: Other possible scenario is to fetch it from within the
+    # LaunchConfiguration
+    #
+    if not ips:
+        return create_kms_data_key()
```

Comment:
Could someone comment on this/review? My hypothesis here is that if get_instance_ips() doesn't return any IPs, we generate a new data_key anyway.